src/config.js

```diff
@@ -2,23 +2,89 @@
  * #### Import members from **@edx/frontend-platform**
  *
  * The configuration module provides utilities for working with an application's configuration
- * document (ConfigDocument).  This module uses `process.env` to import configuration variables
- * from the command-line build process.  It can be dynamically extended at run-time using a
- * `config` initialization handler.  Please see the Initialization documentation for more
- * information on handlers and initialization phases.
+ * document (ConfigDocument).  Configuration environment variables can be supplied to the
+ * application in four different ways:
+ *
+ * ##### Build-time Environment Variables
```

Comment:
I'm not sure the best way to communicate this in this file, but I think it'd be nice to categorize the configuration methods

* Build-time configuration
  * Environment Variables (intend to deprecate)
  * JavaScript File
    * Object
    * Function returning object
    * Function returning promise
* Runtime configuration

Reply:
+1

Reply:
I'm down with that, yeah!

Reply:
I reorganized it a bit to match.  Resolving!